documents/coredocument.go

```diff
@@ -528,3 +537,22 @@ func (cd *CoreDocument) Signatures() (signatures []coredocumentpb.Signature) {
 
 	return signatures
 }
+
+func populateVersions(cd *coredocumentpb.CoreDocument, prevCD *coredocumentpb.CoreDocument) (err error) {
+	if prevCD != nil {
+		cd.PreviousVersion = prevCD.CurrentVersion
+		cd.CurrentVersion = prevCD.NextVersion
+		cd.CurrentPreimage = prevCD.NextPreimage
+	} else {
```

Comment:
you can avoid the else here if you move the nextImage and Nextverison to the top. That makes code easier and reduce cyclomatic complexity

Reply:
Please confirm, I don't think what you are saying can be done. Since its dealing with current version no next version

Reply:
You can rewrite it as

```
    cd.NextPreimage, cd.NextVersion, err = utils.GenerateHashPair(idSize)
	if err != nil {
		return err
	}

    if prevCD != nil {
		cd.PreviousVersion = prevCD.CurrentVersion
		cd.CurrentVersion = prevCD.NextVersion
		cd.CurrentPreimage = prevCD.NextPreimage
        return nil
	}

    cd.CurrentPreimage, cd.CurrentVersion, err = utils.GenerateHashPair(idSize)
    if err != nil {
		return err
	}
	cd.DocumentIdentifier = cd.CurrentVersion
	return nil
}
```

Reply:
The cyclomatci complexity in this case would 3 instead of 4

Reply:
I think I would go with cyclomatic complexity of 4 than creating extra hashes :)